src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1041,10 +1041,6 @@ uint32 Unit::DealDamage(Unit* attacker, Unit* victim, uint32 damage, CleanDamage
 
         if (!victim->IsPlayer())
         {
-            /// @fix: Hack to avoid premature leashing
-            if (damagetype != DOT && damage > 0 && !victim->GetOwnerGUID().IsPlayer() && (!spellProto || !spellProto->HasAura(SPELL_AURA_DAMAGE_SHIELD)))
-                victim->ToCreature()->UpdateLeashExtensionTime();
```

Comment:
Okay, I get it.

Reply:
The leash needs to be extended whenever a spell hits a creature or the creature gets hit with a melee attack. I put an extension in this function in case it helped with the issues you and others were seeing, but in my opinion it shouldn't be there. I was under the impression it served its purpose inside `SetInCombatWith` [here](https://github.com/azerothcore/azerothcore-wotlk/commit/dd2167b00b138e97ff71cf04b54e2de147bb7089#diff-b4a23722c5b449d326b1c0e5a75f3b51de931e42e688793fd9f1d233536a02c6L13545), and my testing seemed to agree with it.